tools/fs_validation.py

```diff
@@ -0,0 +1,281 @@
+    "sociodemo_prob_ageGroup_18_24",
+    "sociodemo_prob_ageGroup_25_34",
+    "sociodemo_prob_ageGroup_35_44",
+    "sociodemo_prob_ageGroup_45_54",
+    "sociodemo_prob_ageGroup_55_64",
+    "sociodemo_prob_ageGroup_65_100",
+    "sociodemo_prob_gender_female",
+    "sociodemo_prob_gender_male",
+    "lookalike_target_DMP_aa9bqvteomrl_cpex",
+    "lookalike_target_DMP_aa9bom9sdj6m_cpex",
+    "lookalike_prob_DMP_aa9bqvteomrl_cpex",
+    "lookalike_prob_DMP_aa9bom9sdj6m_cpex",
+    "income_model_score_low",
+    "income_model_score_mid",
+    "income_model_score_high",
+    "edu_model_score_zs",
+    "edu_model_score_ss_no",
+    "edu_model_score_ss_yes",
+    "edu_model_score_vs",
+    "A_score",
+    "B_score",
+    "C_score",
+    "D_score",
+    "E_score",
+]
+
+# COMMAND ----------
+
+columns_with_upper_case = [
+    "A_score",
+    "B_score",
+    "C_score",
+    "D_score",
+    "E_score",
+    "A_perc",
+    "B_perc",
+    "C_perc",
+    "D_perc",
+    "E_perc",
+]
+
+# COMMAND ----------
+
+# MAGIC %md Helper functions
+
+# COMMAND ----------
+
+
+def _get_columns_to_compare_non_fs(df: DataFrame) -> []:
+    return [col_name for col_name in df.columns if "final" in col_name]
+
+
+# COMMAND ----------
+
+
+def _get_columns_to_compare(df_1: DataFrame, df_2: DataFrame) -> []:
+    words = ["score", "perc", "prob"]
+    # words = ["perc", "prob"]
+    column_list = []
+    df_1_columns = [item.lower() for item in df_1.columns]
+    df_2_columns = [item.lower() for item in df_2.columns]
+    for col_name in df_1_columns:
+        for word in words:
+            if word in col_name and col_name in df_2_columns:
+                column_list.append(col_name)
+    return column_list
+
+
+# COMMAND ----------
+
+
+def _fix_map_datatype(df: DataFrame) -> DataFrame:
+    for column in complex_columns:
+        # specially 'sociodemo', 'score' and 'model' columns
+        if (
+            dict(df.dtypes)[column] == "map<tinyint,double>"
+            or dict(df.dtypes)[column] == "map<tinyint,float>"
+        ):
+            df = df.withColumn(column, F.col(column).getItem(0))
+    return df
+
+
+# COMMAND ----------
+
+
+def _rename_columns_to_lower_case(df: DataFrame) -> DataFrame:
+    for column in columns_with_upper_case:
+        if column in df.columns:
+            df = df.withColumnRenamed(column, column.lower())
+
+    return df
+
+
+# COMMAND ----------
+
+# MAGIC %md Main function
+
+# COMMAND ----------
+
+
+def validate_data_between_two_sources(
+    df_name: str,
+    df_path_old_pipeline: str,
+    coef: float = 0.1,
+    limit_of_rows: int = 10000,
+) -> tuple():
+    """
+    Function that should be able to validate if there are any discrepancies between last
+     two versions of calculated Feature Store rows - bigger than specific thrash-hold
+    Parameters:
+    df_name: Name of table that is specified in config file
+    coef: Thrash-hold under which we expect that rows are calculated just fine
```

Comment:
Its threshold :wink: 
```suggestion
    coef: threshold under which we expect that rows are calculated just fine
```

Reply:
Not sure how I misspelled that 🤣